### ISSUE TITLE
US-maintenance of secure radio connection (EVC - RBC)

### ISSUE THREAD

**Issue by BaseliyosJacob** [openETCS/modeling#397]

**Comment by UweSteinkeFromSiemens** [comment]
How does this relate to Subset 026, chapt. 3.5, "Management of Radio Communication"?

**Comment by stefan-karg** [comment]
The user story has some implications on both MoRC and the checks performed at the input channel, see 3.16.3.4.

According to the chapter 3.16.3.4, the system has to monitor, if radio messages are received in a certain time frame. Otherwise a defined reaction has to be triggered.

If I read chapter 3.5 correctly, then MoRC is monitoring the session on a lower level (since it monitors the Euroradio modules), while the check described in chapter 3.16.3.4 is performed on a higher protocol level.

**Comment by UweSteinkeFromSiemens** [comment]
Indeed, the checks as specified in 3.16.3 pretend to be performed on application level. Due to dependencies with the handing over process, radio hole etc. a clearly layered protocol stack with separate information for each of the layers seems questionable.

**Comment by UweSteinkeFromSiemens** [comment]
Fyi: #430.

**Comment by stefan-karg** [comment]
The part of the check, which checks the timeout (a new radio message is received, before the time specified in T_NVCONTACT is over) was integrated in commit d9057972 . The module "CheckEuroradioMessage" now constantly monitors the reception of new radio messages and signals an error in case the timeout is reached.

**Comment by BaseliyosJacob** [comment]
Allocation of the operation in the OBU System should be discussed.

**Comment by BaseliyosJacob** [comment]
- [x] Analysis
- [x] Modelling
- [ ] Documentation
- [ ] Handover to V&V
- [ ] Approval from V&V

**Comment by stefan-karg** [comment]
Result of discussion with Uwe on July 15: The continous check should not be allocated in the input channel but in the radio management module.

**Comment by BerndHekele** [comment]
@UweSteinkeFromSiemens you please make a proposal  on how to proceed?

**Comment by UweSteinkeFromSiemens** [comment]
@stefan-karg: from my point of view the check would be more appropiate assigned to the protocol's stack safety layer below ATP application level. Nevertheless, if 3.16.3.4 requires such checks to be implemented on application level, this cannot be ignored.

**Comment by BaseliyosJacob** [comment]
Maintenance of radio communication can be tested on the applicatino layer.

**Comment by BaseliyosJacob** [comment]
Should be moved Management of Radio Procedures

**Comment by BaseliyosJacob** [comment]
Is located at the moment in the input channel and needs to be allocated to another location/function.

**Comment by BaseliyosJacob** [comment]
Assigned to @marcbehrens

**Comment by BaseliyosJacob** [comment]
Assigned to @marcbehrens in Sprint 37/38 for V&V

**Comment by MarcBehrens** [comment]
@stefan-karg I started to look at your model and have some questions:

Related to the [ADD document](https://github.com/openETCS/modeling/raw/bd3880c3ed3f5f4553ce181c6765432d328c5ae4/openETCS%20ArchitectureAndDesign/D3.5.3/sections/trainData_V2.tex) Ch 4.1.3.3.1 

1. [clarification] Within row 'Description' please specify exactly what is meant by: 'several checks'.
2. [clarification] Within row 'Description' please describe the the criteria for 'Invalid messages'. 
3. [clarification] Within row 'Description' please describe if the information 'invalid message' is generated within the operator for 'CheckEuroradio Message'

Within the model:

1. I found the variable **CheckEuroradioMessage::cTTrainUnknown1/**  of type real.
Is there any reason why a real is used here?

**Comment by MarcBehrens** [comment]
according to the packet **Obu_BasicTypes_Pkg::T_internal_Type/** the T_internal_Type of tpye INT reads:
' Standardized system time type used for all internal time calculations: in ms' 
@stefan-karg do you agree that this type should be used inside the **checkeuroradiomessage** module?

**Comment by MarcBehrens** [comment]
CheckEuroradioMessage::cTTrainUnknown1/  is related to T_TRAIN subset-026-7.scade which is envisaged to be changes within the next iteration @uwesteinkefromsiemens
